src/components/global/Playground/index.tsx

```diff
@@ -158,21 +170,66 @@ export default function Playground({
 
   const { isDarkTheme } = useThemeContext();
 
+  /**
+   * When deploying, Docusaurus builds the app in an SSR environment.
+   * We need to check whether we're in a browser so we know if we can
+   * use the window or localStorage objects.
+   */
+  const isBrowser = useIsBrowser();
+
   const hostRef = useRef<HTMLDivElement | null>(null);
   const codeRef = useRef(null);
   const frameiOS = useRef<HTMLIFrameElement | null>(null);
   const frameMD = useRef<HTMLIFrameElement | null>(null);
   const consoleBodyRef = useRef<HTMLDivElement | null>(null);
 
-  const defaultMode = typeof mode !== 'undefined' ? mode : Mode.iOS;
+  const { blockElementScrollPositionUntilNextRender } = useScrollPositionBlocker();
```

Comment:
`PlaygroundTabs` already uses this function. Does it need to be included in this file as well?

Reply:
Yes, you have the call the function for every component that needs to block scroll on click.

Reply:
Ah, I see where the confusion might lie -- PlaygroundTabs is for the tabs that switch which file is being shown within the code. The new addition is for the framework switcher for the whole playground.

Reply:
Is it possible to not use `useScrollPositionBlocker`? I'm asking in preparation of the Docusaurus update. It was changed for [internal use only](https://github.com/ionic-team/ionic-docs/pull/3014#discussion_r1281143661). If it's needed, then I'll make a note of it for the update.

Reply:
Hmm 🤔 We definitely need it or something like it; otherwise the scroll can jump around when toggling the frameworks which leads to an unpleasant UX. If we got rid of the import, we'd either need to copy the code directly into our repo, or roll our own solution.

Reply:
Just adding the decision: `useScrollPositionBlocker` will be used as normal even when upgrading Docusaurus.